pkg/metrics/server/server.go

```diff
@@ -249,7 +252,11 @@ func (s *defaultServer) Start(ctx context.Context) error {
 
 func (s *defaultServer) createListener(ctx context.Context, log logr.Logger) (net.Listener, error) {
 	if !s.options.SecureServing {
-		return net.Listen("tcp", s.options.BindAddress)
+		if s.options.ListenConfig == nil {
+			return net.Listen("tcp", s.options.BindAddress)
+		} else {
+			return s.options.ListenConfig.Listen(context.Background(), "tcp", s.options.BindAddress)
```

Comment:
ctx is available, don't use background

Reply:
Yeah I was just trying not to change any behavior for the caller but if you think that is better I will change.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/2519#discussion_r1349817547